### ISSUE TITLE
Question: Changing runtime to 1.0.0 Beta 5 results in error

### ISSUE THREAD

**Issue by mbharodia** [aspnet/dnx#2221]
Hi,
I am running a basic web site created from VS 2015 template. When I installed VS it came with 1.0.0 beta 4. I added the latest runtime 1.0.0 beta 5. The website works with beta4 but not with beta 5. I get the following error 

System.InvalidOperationException: No service for type 'Microsoft.Framework.Runtime.IApplicationEnvironment' has been registered.

I thought that beta5 would be backward compatible with be beta4. So whatever works on beta4 should work on beta5 too. 

Am I missing anything? Thanks in advance for your help.

**Comment by davidfowl** [comment]
> I thought that beta5 would be backward compatible with be beta4. So whatever works on beta4 should work on beta5 too.

This is a bad assumptions. Our betas are not backwards compatible. You need to update everything to beta5 (including packages and you should read the Announcements repo to see the changes https://github.com/aspnet/Announcements/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0-beta5

**Comment by mbharodia** [comment]
Ok. Good to know. I have looked into announcements and could not find a fix to the following error or any other relevant information that may help

System.InvalidOperationException: No service for type 'Microsoft.Framework.Runtime.IApplicationEnvironment' has been registered.

The application does not hit even StartUp metod in webApplication project. I have just created a project using VS 2015 template. Nothing has changes in it except added a global.json file and pinned beta5 runtime. I am using CLR (no core version) version and x64 runtime (1.0.0-beta5). I am running my VS 2015 on Windows 10 (insider preview)

Would you have any idea? Thanks in advance for your help.

**Comment by mbharodia** [comment]
Hi David,

Ok. In that case how do I resolve this error?

I have looked into announcements but was not able find anything relevant to this error.

Actually this error has been thrown even before hitting the StratUp method in Web Application project. As said before I have created a project using VS 2015 web application template and nothing has been modified in it except adding global.json and pinned the version of 1.0.0-beta5 (x64 CLR).

System.InvalidOperationException: No service for type 'Microsoft.Framework.Runtime.IApplicationEnvironment' has been registered.

Is there any idea?

Thanks in advance for your help.
